### ISSUE TITLE
No gradients provided for any variable

### ISSUE THREAD

**Issue by TryArie** [tensorflow/tensorflow#39611]
I'm currently coding a text generation program to compare GRU, LSTM, and Bi-LSTM.

Right now, I'm facing a problem with the gradients problem. It said no gradients provided for any variable.

My best guess is in the model or in the fitting, but no ideas. The other is the dataset.

[I'm using Google CoLab for this program](https://colab.research.google.com/drive/1G4eilixtLJdW0c5TkgiSzunsqK2VUXTT?usp=sharing). I share it in viewer mode (unless you want me to change it).

**Comment by TryArie** [comment]
Well, that explains why. I forget to add a loss function.
